torch/csrc/autograd/python_variable.h

```diff
@@ -43,7 +44,10 @@ static inline bool THPVariable_CheckExact(PyObject *obj) {
 
 inline bool THPVariable_Check(PyObject *obj)
 {
-  return THPVariableClass && PyObject_IsInstance(obj, THPVariableClass);
+  const auto result = PyObject_IsInstance(obj, THPVariableClass);
+  if (result == -1)
+      throw python_error();
+  return THPVariableClass && result;
```

Comment:
The `&&` here to guard the isinstance check and not give it an invalid `THPVariableClass`. This should be preserved and this function should juste return false (without running the isinstance check at all) if `THPVariableClass` is nullptr.

Reply:
@albanD Yes, right. Just fixed!